pool/allocator.go

```diff
@@ -14,22 +14,20 @@ import (
 
 // Factory is responsible for wrapping given command into tasks WorkerProcess.
 type Factory interface {
-	// SpawnWorkerWithTimeout creates a new WorkerProcess process based on given command with context.
+	// SpawnWorkerWithContext creates a new WorkerProcess process based on given command with context.
 	// Process must not be started.
-	SpawnWorkerWithTimeout(context.Context, *exec.Cmd) (*worker.Process, error)
-	// SpawnWorker creates a new WorkerProcess process based on given command.
-	// Process must not be started.
-	SpawnWorker(*exec.Cmd) (*worker.Process, error)
+	SpawnWorkerWithContext(context.Context, *exec.Cmd, ...worker.Options) (*worker.Process, error)
 	// Close the factory and underlying connections.
 	Close() error
 }
 
 // NewPoolAllocator initializes allocator of the workers
-func NewPoolAllocator(ctx context.Context, timeout time.Duration, factory Factory, cmd Command, command []string, log *zap.Logger) func() (*worker.Process, error) {
+func NewPoolAllocator(ctx context.Context, timeout time.Duration, maxJobs uint64, factory Factory, cmd Command, command []string, log *zap.Logger) func() (*worker.Process, error) {
```

Comment:
Internally, we can use `maxExecs`. So, `maxJobs` -> `maxExecs`.

Reply:
Done